### ISSUE TITLE
Onigtest fails

### ISSUE THREAD

**Issue by ScoreUnder** [stedolan/jq#855]
I have no idea how to interpret this, so here's the output:

From setup.sh:
```
PASS: tests/mantest
PASS: tests/jqtest
FAIL: tests/onigtest
PASS: tests/shtest
============================================================================
Testsuite summary for jq 1.5rc1-153-g54add5a
============================================================================
# TOTAL: 4
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
Please report to mu@netsoc.tcd.ie
============================================================================
Makefile:1089: recipe for target 'test-suite.log' failed
make[3]: *** [test-suite.log] Error 1
make[3]: Leaving directory '/home/score/src/jq'
Makefile:1195: recipe for target 'check-TESTS' failed
make[2]: *** [check-TESTS] Error 2
make[2]: Leaving directory '/home/score/src/jq'
Makefile:1425: recipe for target 'check-am' failed
make[1]: *** [check-am] Error 2
make[1]: Leaving directory '/home/score/src/jq'
Makefile:1427: recipe for target 'check' failed
make: *** [check] Error 2
make  install-am
make[1]: Entering directory '/home/score/src/jq'
make[2]: Entering directory '/home/score/src/jq'
 /usr/bin/mkdir -p './tmp/opt/junk/lib'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libjq.la './tmp/opt/junk/lib'
Usage: /home/score/src/jq/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: './tmp/opt/junk/lib' must be an absolute directory name
Makefile:771: recipe for target 'install-libLTLIBRARIES' failed
make[2]: *** [install-libLTLIBRARIES] Error 1
make[2]: Leaving directory '/home/score/src/jq'
Makefile:1442: recipe for target 'install-am' failed
make[1]: *** [install-am] Error 2
make[1]: Leaving directory '/home/score/src/jq'
Makefile:1436: recipe for target 'install' failed
make: *** [install] Error 2
make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/home/score/src/jq'
make[1]: *** No rule to make target 'tests/modules/streaming.jq', needed by 'distdir'.  Stop.
make[1]: Leaving directory '/home/score/src/jq'
Makefile:1336: recipe for target 'dist' failed
make: *** [dist] Error 2
./setup.sh  131.30s user 5.41s system 96% cpu 2:22.34 total
```

[test-suite.log](http://paste.pound-python.org/show/EETO82BgWlrEbZls7qYz/)

**Comment by ScoreUnder** [comment]
```
checking whether yytext is a pointer... yes
./configure: 14111: test: 0: unexpected operator
checking oniguruma.h usability... yes
checking oniguruma.h presence... yes
```
I think this might have something to do with it.

`if test $ONIGURUMAPATHSET == 1; then`

That line should read:
`if test "$ONIGURUMAPATHSET" = 1; then`

The `test` command has no `==` operator. `=` is used for testing string equality, but some shells (bash) add `==` for some bizarre reason.

...but, fixing it doesn't fix the test :(

**Comment by dtolnay** [comment]
Thanks for reporting this. It looks like Nico fixed the "=" in 8127031, which I think should have fixed the onigtest failures in the test-suite.log you attached given that Autoconf can find oniguruma.h. Your second failure is in the `make install` step which should have used an absolute path instead of `./tmp`. I have removed that step in ceb2684 because it should be enough for setup.sh to run `make check`.

Please try setup.sh again and let me know if it is still failing.

**Comment by ScoreUnder** [comment]
`setup.sh` works well for me now - both errors mentioned in this thread don't happen any more.
Thanks for the fix.
